src/main/java/reo/Storage.java

```diff
@@ -0,0 +1,177 @@
+package reo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/** Supports the required file operations for reo.Reo. */
+public class Storage {
+    private String filePath;
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+    /**
+     * Returns true if the parameter is 1, and false otherwise.
+     *
+     * @param i The integer used to represent true or false in the saved file.
+     * @return Boolean representation of i.
+     */
+    public static boolean convertIntToBool(int i) {
+        return i == 1;
+    }
+
+    /**
+     * Reads the saved file, and interprets its contents to create
+     * an ArrayList of reo.Task objects represented by the file contents.
+     *
+     * @return An ArrayList of Tasks interpreted from the file.
+     */
+    public ArrayList<Task> readFile() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File dir = new File("./data");
+            if (!dir.exists()) {
+                dir.mkdir();
+                System.out.println("Created new directory to store data.");
+            }
+            File f = new File(filePath);
+            if (f.createNewFile()) {
+                System.out.println("Created new file to store list.");
+                return tasks;
+            }
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                String[] split = line.split("\\|");
+
+                switch (split[0].trim()) {
+                    case "T":
+                        boolean isDoneT = convertIntToBool(Integer.parseInt(split[1].trim()));
+                        String nameT = split[2].trim();
+                        tasks.add(new Todo(nameT, isDoneT));
+                        break;
+                    case "E":
+                        boolean isDoneE = convertIntToBool(Integer.parseInt(split[1].trim()));
+                        String nameE = split[2].trim();
+                        String from = split[3].trim();
+                        String to = split[4].trim();
+                        tasks.add(new Event(nameE, isDoneE, to, from));
+                        break;
+                    case "D":
+                        boolean isDoneD = convertIntToBool(Integer.parseInt(split[1].trim()));
+                        String nameD = split[2].trim();
+                        String deadline = split[3].trim();
+                        tasks.add(new Deadline(nameD, isDoneD, deadline));
+                        break;
+                }
```

Comment:
Case clauses should not be indented.
Include the default case to check for invalid inputs and handle the error accordingly.
Try to shorten the method by implementing the changes above.